### ISSUE TITLE
Save model checkpoints during training

### ISSUE THREAD

**Issue by riadsouissi** [Microsoft/LightGBM#1411]
Hi. 
When training takes many hours to a full day, saving the model every number of iterations is extremely useful in case the process is interrupted (I use preemptible google cloud instance which shuts down when there is a high load).
Is it possible to do with or is it a feature request?

**Comment by guolinke** [comment]
@riadsouissi 
we already has this function, refer to parameter `snapshot_freq` .

**Comment by riadsouissi** [comment]
Thank you.
I am not able to find the documentation for this parameter and how to use it. Can you please refer me to it? I use python by the way.

**Comment by riadsouissi** [comment]
Thank you.
I added this parameter along other LGBM parameters like below (last key) but I don't see any model file being created. Where would be the file created and what would it be named ?

{'boosting_type': 'gbdt', 'objective': 'binary', 'metric': ['auc'], 'learning_rate': 0.05, 'num_leaves': 120, 'max_depth': -1, 'min_child_samples': 100, 'max_bin': 1000, 'subsample': 0.9, 'subsample_freq': 1, 'colsample_bytree': 0.6, 'min_child_weight': 30, 'subsample_for_bin': 200000,'reg_alpha': 0, 'reg_lambda': 0, 'nthread': -1, 'verbose': 0, 'scale_pos_weight': 1, 'snapshot_freq': 5}

**Comment by guolinke** [comment]
@riadsouissi this function only can be used in cli version. 

For python-package, you can use use callback function to achieve saving models.

**Comment by yangtian62** [comment]
Hi Mr. Ke, I set `snapshot_freq = 10` and `output_model = /path/to/model.txt` in my trian.conf but still no models saved. I'm using MPI parallel version. Is there something I missed? Or does parallel learning support this func?

**Comment by guolinke** [comment]
@yangtian62 
I just check the code, it should work in both serial and parallel versions.
Can you try the serial version?

**Comment by yangtian62** [comment]
It was due to the permission issue. I changed another cluster and it worked fine. Thanks! @guolinke
